frontend/src/scenes/settings/Settings.tsx

```diff
@@ -80,41 +83,53 @@ export function Settings({
                   </OptionButton>
               ),
           }))
-        : levels.map((level) => ({
-              key: level,
-              content: (
-                  <OptionButton
-                      to={urls.settings(level)}
-                      handleLocally={handleLocally}
-                      active={selectedLevel === level && !selectedSectionId}
-                      onClick={() => selectLevel(level)}
-                  >
-                      <span className="text-secondary">{SettingLevelNames[level]}</span>
-                  </OptionButton>
-              ),
-              items: sections
-                  .filter((x) => x.level === level)
-                  .map((section) => ({
-                      key: section.id,
-                      content: (
-                          <OptionButton
-                              to={section.to ?? urls.settings(section.id)}
-                              handleLocally={handleLocally}
-                              active={selectedSectionId === section.id}
-                              isLink={!!section.to}
-                              onClick={() => {
-                                  if (section.to) {
-                                      router.actions.push(section.to)
-                                  } else {
-                                      selectSection(section.id, level)
-                                  }
-                              }}
-                          >
-                              {section.title}
-                          </OptionButton>
-                      ),
-                  })),
-          }))
+        : filteredLevels.map((level) => {
+              const levelSections = filteredSections.filter((x) => x.level === level)
+              const isCollapsed = collapsedLevels[level]
+              const hasItems = levelSections.length > 0
+
+              return {
+                  key: level,
+                  content: (
+                      <OptionButton
+                          handleLocally={handleLocally}
+                          active={selectedLevel === level && !selectedSectionId}
+                          onClick={() => {
+                              if (hasItems) {
+                                  toggleLevelCollapse(level)
+                              } else {
+                                  selectLevel(level)
+                              }
+                          }}
+                          sideIcon={hasItems ? isCollapsed ? <IconChevronRight /> : <IconChevronDown /> : undefined}
```

Comment:
I agree that this is hard to read, there's an easy-ish fix
```suggestion
                          sideIcon={hasItems && (isCollapsed ? <IconChevronRight /> : <IconChevronDown />)}
```